org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -397,6 +408,25 @@ private Range toReplacementRange(CompletionProposal proposal){
 		return null;
 	}
 
+	private void setInsertReplaceRange(CompletionProposal proposal, InsertReplaceEdit edit){
+		try {
+			int start = proposal.getReplaceStart();
+			int end = proposal.getReplaceEnd();
+			if (edit.getReplace() == null) {
+				Range replaceRange = JDTUtils.toRange(compilationUnit, start, end - start);
+				edit.setReplace(replaceRange);
+			}
+
+			if (edit.getInsert() == null) {
+				end = Math.min(end, offset);
```

Comment:
Is there a situation where the proposal's end offset is smaller than the cursor offset ?

Reply:
I didn't find a case for that so far. Since the current implementation does the same thing: https://github.com/eclipse/eclipse.jdt.ls/blob/e3f5c743fdb395a8d1b6ef5154c75645b5939380/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java#L164-L169.

It should be fine otherwise we should receive users' feedback for that.